src/components/ConfigEditor/ConnectionConfig.tsx

```diff
@@ -164,33 +165,36 @@ const ConnectionConfig: React.FC<ConnectionConfigProps> = ({
         tooltip={clientSecretTooltip}
       />
 
-      <InlineField
-        label="Use On-Behalf-Of"
-        labelWidth={26}
-        tooltip={
-          <>
-            Propagate Grafana client credentials to ADX with a token exchange. When enabled the service account (Client
-            ID) impersonates the user by augmenting the access token. See the{' '}
-            <a
-              target="_blank"
-              rel="noreferrer"
-              href="https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow"
-            >
-              developer documentation
-            </a>{' '}
-            for detail on the concept. This feature requires a Grafana version 8.3.4 or later.
-          </>
-        }
-      >
-        <HorizontalGroup>
-          <InlineSwitch
-            id="adx-on-behalf-of"
-            value={jsonData.onBehalfOf}
-            onChange={(ev: React.ChangeEvent<HTMLInputElement>) => updateJsonData('onBehalfOf', ev.target.checked)}
-          />
-          <span>⚠️ This feature is in beta</span>
-        </HorizontalGroup>
-      </InlineField>
+      {config.featureToggles.adxOnBehalfOf && (
```

Comment:
I would also add a check in the first useEffect her (line 30) so in case an existing data source has the feature enabled but it's disabled by configuration it updates the value:

```ts
    if (jsonData.onBehalfOf && !config.featureToggles.adxOnBehalfOf) {
      updateJsonData('onBehalfOf', false);
    }
```